pynetdicom/dimse_messages.py

```diff
@@ -310,14 +319,50 @@ def decode_msg(self, primitive):
                         #   has been completely decoded
                         return True
 
+                    if (
+                        self.__class__.__name__ == "C_STORE_RQ"
+                        and _config.STORE_RECV_CHUNKED_DATASET
+                    ):
+                        self._data_set_base_dir = TemporaryDirectory()
+                        self._data_set_path = (
+                            Path(self._data_set_base_dir.name)
+                            / f"{uuid.uuid4()}.dcm"
+                        )
+
+                        with self._data_set_path.open("wb") as data_set_path:
+                            from pynetdicom import PYNETDICOM_IMPLEMENTATION_UID
+                            from pynetdicom import PYNETDICOM_IMPLEMENTATION_VERSION
+
+                            data_set_path.write(b'\x00' * 128)
+                            data_set_path.write(b'DICM')
+
+                            file_meta = Dataset()
+
+                            file_meta.FileMetaInformationGroupLength = 0
+                            file_meta.FileMetaInformationVersion = b'\x00\x01'
+                            file_meta.MediaStorageSOPClassUID = self.command_set.AffectedSOPClassUID
+                            file_meta.MediaStorageSOPInstanceUID = self.command_set.AffectedSOPInstanceUID
+                            file_meta.TransferSyntaxUID = assoc._accepted_cx[context_id].transfer_syntax[0]
+                            file_meta.ImplementationClassUID = PYNETDICOM_IMPLEMENTATION_UID
+                            file_meta.ImplementationVersionName = PYNETDICOM_IMPLEMENTATION_VERSION
+
+                            file_meta.is_little_endian = True
+                            file_meta.is_implicit_VR = False
```

Comment:
This largely duplicates the logic in the file_meta property on the event - if you'd like I can pull out a helper that produces this

Reply:
Yeah if you want, otherwise I can do it. It should probably go in `dsutils`.